python/extism/extism.py

```diff
@@ -105,47 +107,107 @@ def _wasm(plugin):
     return wasm
 
 
+class Context:
```

Comment:
Minor note:

Perhaps this could also support the context manager dunder methods since it semantically maps well to the `with` language feature: https://docs.python.org/3/library/contextlib.html